linenoise.c

```diff
@@ -1168,7 +1170,11 @@ int linenoiseHistorySave(const char *filename) {
     fp = fopen(filename,"w");
     umask(old_umask);
     if (fp == NULL) return -1;
-    chmod(filename,S_IRUSR|S_IWUSR);
+    if (chmod(filename,S_IRUSR|S_IWUSR) < 0) {
```

Comment:
This was the fix for the history file saving issue:

https://github.com/antirez/linenoise/commit/c894b9e59f02203dbe4e2be657572cf88c4230c3